src/main/java/org/opensearch/sdk/ExtensionSettings.java

```diff
@@ -19,30 +19,26 @@ public class ExtensionSettings {
     // Change the location to extension.yml file of the extension
     public static final String EXTENSION_DESCRIPTOR = "src/test/resources/extension.yml";
 
-    public String getExtensionname() {
-        return extensionname;
+    public ExtensionSettings(String extensionname, String hostaddress, String hostport) {
```

Comment:
I generally like the use of a constructor like this to set the fields, but this also implies these will always be the only fields we'll be using.  If we add one more we'd have to create a 4-arg constructor but because of SemVer we couldn't delete the 3-arg one; we could deprecate it but then we'd have to deal with what the appropriate default is etc.

All this is to say that this is only a good idea if we are absolutely sure these are the only 3 fields we will ever likely have in this class.

Reply:
Actually, this is the way of initialization required by ObjectMapper to load yml file and the other was using setters but since we don't want to expose or modify host address and port, this is the approach I went for. Open to suggestions though.

Reply:
Hard-coding inputs (as in constructors) is nearly always disconnected both logically and programmatically from reading in an arbitrary number of configurations key/value pairs via an external file. 

This is an interesting use case in that we want a POJO which is what we currently have but we also want to reserve the right to later generate a different POJO with more fields.

I took a look at ObjectMapper docs and it seems that it uses reflection and you can actually use private constructors for this purpose. That may help settle the SemVer argument as it is obvious non-API.  Alternately if there is some way we can mark this class as "internal" (a package with internal in the name is the common approach) then we also aren't bound by SemVer.

So the approach may be correct but the file location may not be. We need special treatment (package location) for these POJO objects.

Reply:
As discussed. Removed the arg constructor and called super for no arg constructor